### DISCUSSION TITLE
JSI error when building application

### DISCUSSION THREAD

**Discussion by GorbunovV** [jpudysz/react-native-unistyles#257] [Q&A] [answered]
Hello. Maybe the community can help me.

When updating react-native-unistyles from version 2.7.2 to version 2.9.0, I encountered a problem. The application stopped building on the Android and iOS platforms with the same error. Below I will show the essence of the problem with screenshots.

I'm using the version react-native = 0.72.15

Perhaps version 2.9.0 does not support react-native versions lower than 0.73.*? Although the release notes say backwards compatibility.

{path} = hidden path to folder

**Android**

> In file included from {path}/mobile-native/node_modules/react-native-unistyles/cxx/UnistylesImpl.cpp:1:
> In file included from{path}/mobile-native/node_modules/react-native-unistyles/android/../cxx/UnistylesRuntime.h:3:
> In file included from {path}/node_modules/react-native-unistyles/android/../cxx/UnistylesModel.h:3:
> 
> {path}/.gradle/caches/transforms-3/51ace0538d305a17b48524f84af1d856/transformed/jetified-react-android-0.72.15-debug/prefab/modules/jsi/include/jsi/jsi.h:1102:43: error: no matching function for call to 'kindOf'
> 
> {path}/mobile-native/node_modules/react-native-unistyles/cxx/UnistylesImpl.cpp:237:12: note: in instantiation of function template specialization 'facebook::jsi::Value::Value<float>' requested here
> 
> {path}/.gradle/caches/transforms-3/51ace0538d305a17b48524f84af1d856/transformed/jetified-react-android-0.72.15-debug/prefab/modules/jsi/include/jsi/jsi.h:1350:30: note: candidate function not viable: no known conversion from 'float' to 'const facebook::jsi::Symbol' for 1st argument
> 
> {path}/.gradle/caches/transforms-3/51ace0538d305a17b48524f84af1d856/transformed/jetified-react-android-0.72.15-debug/prefab/modules/jsi/include/jsi/jsi.h:1353:30: note: candidate function not viable: no known conversion from 'float' to 'const facebook::jsi::BigInt' for 1st argument
> 
> {path}/.gradle/caches/transforms-3/51ace0538d305a17b48524f84af1d856/transformed/jetified-react-android-0.72.15-debug/prefab/modules/jsi/include/jsi/jsi.h:1356:30: note: candidate function not viable: no known conversion from 'float' to 'const facebook::jsi::String' for 1st argument
> 
> {path}/.gradle/caches/transforms-3/51ace0538d305a17b48524f84af1d856/transformed/jetified-react-android-0.72.15-debug/prefab/modules/jsi/include/jsi/jsi.h:1359:30: note: candidate function not viable: no known conversion from 'float' to 'const facebook::jsi::Object' for 1st argument
> 
> /{path}/.gradle/caches/transforms-3/51ace0538d305a17b48524f84af1d856/transformed/jetified-react-android-0.72.15-debug/prefab/modules/jsi/include/jsi/jsi.h:1103:5: error: static_assert failed due to requirement 'std::is_base_of<facebook::jsi::Symbol, float>::value || std::is_base_of<facebook::jsi::BigInt, float>::value || std::is_base_of<facebook::jsi::String, float>::value || std::is_base_of<facebook::jsi::Object, float>::value' "Value cannot be implicitly move-constructed from this type"

**IOS**

> {path}/mobile-native/node_modules/react-native-unistyles/cxx/UnistylesImpl.cpp:1:10: in file included from {path}mobile-native/node_modules/react-native-unistyles/cxx/UnistylesImpl.cpp:1:
> 
> {path}/mobile-native/node_modules/react-native-unistyles/cxx/UnistylesRuntime.h:3:10: in file included from {path}/mobile-native/node_modules/react-native-unistyles/cxx/UnistylesRuntime.h:3:
> 
> {path}/mobile-native/node_modules/react-native-unistyles/cxx/UnistylesModel.h:3:10: in file included from {path}/mobile-native/node_modules/react-native-unistyles/cxx/UnistylesModel.h:3:
> 
> {path}/mobile-native/node_modules/react-native-unistyles/cxx/UnistylesImpl.cpp:237:12: In instantiation of function template specialization 'facebook::jsi::Value::Value<float>' requested here
> 
> {path}/mobile-native/ios/Pods/Headers/Public/hermes-engine/jsi/jsi.h:1350:30: Candidate function not viable: no known conversion from 'float' to 'const Symbol' for 1st argument
> 
> {path}/mobile-native/ios/Pods/Headers/Public/hermes-engine/jsi/jsi.h:1353:30: Candidate function not viable: no known conversion from 'float' to 'const BigInt' for 1st argument
> 
> {path}/mobile-native/ios/Pods/Headers/Public/hermes-engine/jsi/jsi.h:1356:30: Candidate function not viable: no known conversion from 'float' to 'const String' for 1st argument
> 
> {path}/mobile-native/ios/Pods/Headers/Public/hermes-engine/jsi/jsi.h:1359:30: Candidate function not viable: no known conversion from 'float' to 'const Object' for 1st argument

**Comment by jpudysz** [comment, marked as answer]
Hey 👋
Unistyles >= 2.8.0 requires min React Native 0.73.0.

Next version will support only React Native 0.75. It's because a lot of changes in the core of RN.
